packages/types/src/joiner/index.ts

```diff
@@ -0,0 +1,31 @@
+export type JoinerRelationship = {
+  alias: string
+  foreignKey: string
+  primaryKey: string
+  serviceName: string
+  inverse?: boolean // In an inverted relationship the foreign key is on the other service and the primary key is on the current service
+}
+
+export interface JoinerServiceConfig {
+  serviceName: string
+  primaryKeys: string[]
+  relationships: JoinerRelationship[]
+}
+
+export interface JoinerArgument {
+  name: string
+  value?: any
+  field?: string
+}
+
+export interface RemoteJoinerQuery {
```

Comment:
todo: We are missing the order property which usually allow one order such as the following example `-variants.prices.amount` `variants.prices.amount`

Reply:
About Ordering, we will have to have a dedicated discussion around that. I don't want to come up with any rushed implementation to support this.